modules/nf-core/find/unpigz/main.nf

```diff
@@ -0,0 +1,56 @@
+process FIND_UNPIGZ {
+    tag "${meta.id}"
+    label 'process_medium'
+
+    conda "${moduleDir}/environment.yml"
+    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
+        ? 'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/7f/7fd226561e12b32bcacdf4f5ff74577e76233adf52ae5cbc499a2cdfe0e27d82/data'
+        : 'community.wave.seqera.io/library/findutils_pigz:c4dd5edc44402661'}"
+
+    input:
+    tuple val(meta), path(files_in, stageAs: 'gzipped/*')
+
+    output:
+    tuple val(meta), path("${prefix}.*"), emit: file_out
+    path "versions.yml", emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ""
+    prefix = task.ext.prefix ?: "${meta.id}"
+
+    if (files_in.any { file -> !file.name.endsWith('.gz') }) {
+        error("All files provided to this module must be gzipped (and have the .gz extension).")
+    }
+
+    """
+    while IFS= read -r -d \$'\\0' file; do
+        unpigz \\
+            ${args} \\
+            -cd \\
+            --processes ${task.cpus} \\
+            \$file \\
+            > ${prefix}.\$( basename \$file .gz )
+    done < <( find gzipped/ -name '*.gz' -print0 )
```

Comment:
I still wonder if it would be more efficiency to give each pigz job 1 CPU and parallelise the number of gunzip processes with e.g. xargs

Reply:
The issue with that approach is that you can end up with poor resource utilisation if you have many small files but one large